packages/shared/src/components/tooltips/LinkWithTooltip.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { ReactElement, useMemo, useState } from 'react';
+import Link, { LinkProps } from 'next/link';
+import { BaseTooltip, getShouldLoadTooltip } from './BaseTooltip';
+import { SimpleTooltipProps } from './SimpleTooltip';
+
+interface LinkWithTooltipProps extends LinkProps {
+  children?: ReactElement;
+  tooltip: SimpleTooltipProps;
+}
+
+export function LinkWithTooltip({
+  children,
+  tooltip,
+  ...props
+}: LinkWithTooltipProps): ReactElement {
+  const [element, setElement] = useState<Element>();
```

Comment:
Why don't you use an actual `useRef` here?

Reply:
I did before, but an issue arises when it comes to using the `QuaternaryButton`, it could be an existing issue for twisting refs together. I will try to reproduce the case and create an issue at `react` or `nextjs`.

The refs suddenly won't be available so `onclick` never works. But on other components having straightforward implementations of ref, it works.

Reply:
Ok so don't put more effort into it. If you tried it before and it doesn't work that's fine.

Reply:
Thank you! 🚀